### PR TITLE
Creating shared-config for console to consume

### PR DESCRIPTION
**Pull request by ewolinetz**
Addresses https://jira.coreos.com/browse/LOG-233

@spadgett PTAL and confirm this would work. There will always be `kibanaAppHost` and `kibanaInfraHost` entries, even if it is an "all-in-one" configuration where we only create a single `route`.

e.g.
```
oc get configmap -o yaml sharing-config
apiVersion: v1
data:
  kibanaAppHost: kibana-app
  kibanaInfraHost: kibana-infra
kind: ConfigMap
metadata:
  creationTimestamp: 2018-11-30T18:21:05Z
  name: sharing-config
  namespace: openshift-logging
  ownerReferences:
  - apiVersion: logging.openshift.io/v1alpha1
    controller: true
    kind: ClusterLogging
    name: example-cluster-logging
    uid: b27ce418-f4cc-11e8-a9ff-0e3a960e086c
  resourceVersion: "7727"
```

### CODE REVIEW

**Comment by spadgett** [comment]
Do we need to say if it's http/https?

**Comment by spadgett** [comment]
Here is what the monitoring team did if we want to be consistent:

https://github.com/openshift/cluster-monitoring-operator/pull/155
https://github.com/openshift/cluster-monitoring-operator/pull/169
https://github.com/openshift/cluster-monitoring-operator/pull/170

**Comment by ewolinetz** [comment]
@spadgett that is a good question. What did we have before? It would likely be best to avoid regressions from that.

**Comment by spadgett** [comment]
We had the full URL in the config, but we did require https since we passed
the user’s token.

File: pkg/k8shandler/visualization.go [hunk 1]
```diff
@@ -488,3 +507,14 @@ func updateCurrentImages(current *apps.Deployment, desired *apps.Deployment) (*a
 
 	return current
 }
+
+func createSharedConfig(logging *logging.ClusterLogging, kibanaAppHost, kibanaInfraHost string) *v1.ConfigMap {
+	return utils.ConfigMap(
+		"sharing-config",
+		logging.Namespace,
+		map[string]string{
+			"kibanaAppHost":   kibanaAppHost,
```
Comment (richm) [inline pkg/k8shandler/visualization.go]: Is this a host or a url?  The arguments to `createSharedConfig` are urls

Comment (ewolinetz) [reply pkg/k8shandler/visualization.go]: updated name to be more appropriate

**Comment by ewolinetz** [comment]
Also added in missing `TLS` section for route

**Pull request merged**